components/blitz/src/omero/gateway/model/MaskData.java

```diff
@@ -368,5 +368,198 @@ public byte getBit(byte[] data, int bit)
                 (byte)1 : (byte)0);
     }
 
+    /**
+     * Set the mask
+     * 
+     * @param mask
+     *            The binary mask
+     */
+    public void setMask(int[] mask) {
+        if (getWidth() == 0 || getHeight() == 0)
+            throw new IllegalArgumentException(
+                    "This method requires width and height of the shape");
+        
+        int height = (int) getHeight();
+        setMask(fold(intToBoolean(mask), height));
+    }
+
+    /**
+     * Set the mask
+     * 
+     * @param mask
+     *            The binary mask (int[width][height])
+     */
+    public void setMask(int[][] mask) {
+        setMask(intToBoolean(mask));
+    }
+
+    /**
+     * Set the mask
+     * 
+     * @param mask
+     *            The binary mask
+     */
+    public void setMask(boolean[] mask) {
+        if (getWidth() == 0 || getHeight() == 0)
+            throw new IllegalArgumentException(
+                    "This method requires width and height of the shape");
+
+        int height = (int) getHeight();
+        setMask(fold(mask, height));
+    }
 
+    /**
+     * Set the mask
+     * 
+     * @param mask
+     *            The binary mask (boolean[width][height])
+     */
+    public void setMask(boolean[][] mask) {
+        int width = mask.length;
+        int height = mask[0].length;
+        int offset = 0;
+        byte[] data = new byte[(int)(width * height / 8 + 1)];
+        for (int y = (int) height - 1; y >= 0; y--) {
+            for (int x = 0; x < (int) width; x++) {
+                setBit(data, offset, mask[x][y] ? 1 : 0);
+                offset++;
+            }
+        }
+        setMask(data);
+    }
+    
+    /**
+     * Set the mask; automatically crop to its bounding box.
+     * (sets X, Y, width and height accordingly)
+     * Needs the height initially set to the image height.
+     * 
+     * @param mask
+     *            The binary mask covering the whole
+     *            image
+     */
+    public void setMaskAndCrop(int[] mask) {
```

Comment:
Thought about it again... these methods should better go into some kind of utility class not in the MaskData itself. Will remove the `setAndCrop` methods again.